Add login.microsoftonline.com to internal login pages (#1205)
nativefier/nativefier: API.md

nativefier https://google.com --internal-urls ".*?"
```

##### Internal Login Pages

> *[Breaking change in Nativefier 43.0.0]* Finally, URLs for known login pages

Reviewer: MD nit:

```suggestion

*[Breaking change in Nativefier 43.0.0]* Finally, URLs for known login pages
```

are considered internal. This does not replace `internal-urls`, it complements
it, and happens *before* your `internal-urls` rule is applied. So, if you
already set the flag to let such auth pages open internally, you don't need to
change it but it might be unnecessary.

> We think this is desirable behavior and are so far unaware of cases where users
> might not want this. If you disagree, please chime in at
> [PR #1124: App: Automatically consider known login pages as internal](https://github.com/nativefier/nativefier/pull/1124)
> 

Reviewer: We didn't get any negative feedback about this since it landed, time to scrap the call for feedback 🙂

```suggestion
```

> Current known internal login pages:
> - `amazon.com/signin`
> - `facebook.com/login`
> - `github.com/login`
> - `github.com/session`
> - `accounts.google.com`
> - `mail.google.com/accounts/SetOSID`
> - `linkedin.com/uas/login`
> - `login.live.com`
> - `login.microsoftonline.com`
> - `okta.com`
> - `twitter.com/oauth/authenticate`
> - `appleid.apple.com/auth/authorize`

Reviewer: Hmmmm, do we really want to maintain this full list here? What about instead
```suggestion
Example of internal login pages: `amazon.com/signin`, `facebook.com/login`, `github.com/login`, `accounts.google.com` ...
```

Author: > Hmmmm, do we really want to maintain this full list here? What about instead

The reason I did this was specifically around the issue this PR addresses. Namely, we said "Microsoft logins should work", but one didn't. Transparency here could be useful for an end user to know whether what they're seeing is a bug or expected behavior.

Alternatively, I thought about doing away with the list (here and in code) and instead going off "keywords" like `login` or `session ` but I'd be worried about false positives.

Reviewer: > > Hmmmm, do we really want to maintain this full list here? What about instead
> 
> The reason I did this was specifically around the issue this PR addresses. Namely, we said "Microsoft logins should work", but one didn't. Transparency here could be useful for an end user to know whether what they're seeing is a bug or expected behavior.
> 
> Alternatively, I thought about doing away with the list (here and in code) and instead going off "keywords" like `login` or `session ` but I'd be worried about false positives.

@TheCleric makes sense. Suggestion, then: in `isInternalLoginPage`, change

```
// Making changes? Remember to update the tests in helpers.test.ts
```

to

```
// Making changes? Remember to update the tests in helpers.test.ts and API.md / "Internal Login Pages"
```

, and by the way I'd copy-pasta it to the *end* of the list too (in addition to the beginning), for more visibility.

Note: While .com is specified, for most of these we try to match even on non-US
based domains such as `.co.uk` as well

> If you think this list is missing a login page that you think should be internal, feel free to submit an [issue](https://github.com/nativefier/nativefier/issues/new?assignees=&labels=feature-request&template=feature_request.md&title=[Feature%20Request]%20-%20New%20internal%20login%20page%20request) or even better a pull request!

Reviewer: ```suggestion
If you think this list is missing a login page that you think should be internal, feel free to submit an [issue](https://github.com/nativefier/nativefier/issues/new?assignees=&labels=bug&template=bug_report.md&title=[New%20internal%20login%20page%20request]%20Your%20login%20page%20here) or even better a pull request!
```

?
